### PR TITLE
Audio: SRC: Increase stage buffer size and change copy() run condition

### PR DESCRIPTION
**Pull request by singalsu**
The main changes are commits 1 and 2 to SRC code. First prevents a pipeline freeze. The second fixes an overflow with 16 bit audio data.

The rest of changes are improvements to src_test.m script. E.g. call testbench with copy limit of 5 min to stop frozen SRC and minor issues fixes and speedup with reduce of printing and plot window opening.

### CODE REVIEW

File: tools/test/topology/test-playback.m4 [hunk 1]
```diff
@@ -152,8 +152,8 @@ SectionGraph."pipe-sof-fourth-pipe" {
 
 # DAI configuration
 
-# Use 3 periods for SRC DAI buffer, otherwise 2 periods
-ifelse(TEST_PIPE_NAME, `src', `define(TEST_DAI_PERIODS, `3')', `define(TEST_DAI_PERIODS, `2')')
+# Use 4 periods for SRC DAI buffer, otherwise 2 periods
```
Comment (singalsu) [inline tools/test/topology/test-playback.m4:155]: These are TBD, maybe 3 & 3 could work. I'll test. I did this before noticing the stage1 and stage2 conditions.

Comment (singalsu) [reply tools/test/topology/test-playback.m4]: Yes, the existing 3 source and 3 sink buffers were sufficient. I removed this topology commit.

File: tools/test/audio/src_test.m [hunk 2]
```diff
@@ -413,7 +413,9 @@
 %% Analyze
 test.fs = t.fs2;
 test = chirp_test_analyze(test);
-src_test_result_print(t, 'Chirp', 'chirpf');
+if ~test.fail
```
Comment (singalsu) [inline tools/test/audio/src_test.m:416]: Mistake here, the fail is -1 if the chirp file doesn't exist. Need to plot & store a failed chirp spectra.

Comment (singalsu) [reply tools/test/audio/src_test.m]: fixed now

**Review by lgirdwood — APPROVED** [review]
Approve the C part.

**Comment by gongjun-song** [comment]
This PR indeed fix hang issue, but I tested src_test.sh on the default SOF and it didn't pass all the test cases.
```
Number of passed tests = 864
Number of failed tests = 2
Number of non-applicable tests = 156
Number of skipped tests = 0

ERROR: TEST FAILED!!!
Error: SRC test with 16 bits data failed.
Test failed.
```

**Comment by singalsu** [comment]
> This PR indeed fix hang issue, but I tested src_test.sh on the default SOF and it didn't pass all the test cases.

Thanks for report! I ran the 32 bit test. There is likely some pass criteria that the 16 bit version cannot pass, I'll try to address it. Please use meanwhile 24 or 32 bit tests.

**Comment by gongjun-song** [comment]
> > This PR indeed fix hang issue, but I tested src_test.sh on the default SOF and it didn't pass all the test cases.
> 
> Thanks for report! I ran the 32 bit test. There is likely some pass criteria that the 16 bit version cannot pass, I'll try to address it. Please use meanwhile 24 or 32 bit tests.

I disabled 16 bit and only test 24bit and 32bit, all passed.
```
Number of passed tests = 866
Number of failed tests = 0
Number of non-applicable tests = 156
Number of skipped tests = 0

Test passed.
Success: SRC test with 24 and 32 bits data passed.
Test passed.
```

**Comment by singalsu** [comment]
@gongjun-song @lgirdwood The 16 bit issue was a real bug, good catch! I added as 2nd commit the bug fix to polyphase filter where the saturation was missing for 16 bit audio data round.

**Comment by gongjun-song** [comment]
New commits have fixed 16-bit issue, all passed!
```
Number of passed tests = 866
Number of failed tests = 0
Number of non-applicable tests = 156
Number of skipped tests = 0

Test passed.
Success: SRC test with 16, 24 and 32 bits data passed.
Test passed.
```

**Review by lgirdwood — APPROVED** [review]
@gongjun-song thanks for the test results.

**Pull request merged**